lib/src/rules/invariant_booleans.dart

```diff
@@ -182,9 +182,10 @@ class _ContradictionReportRule extends LintRule {
 }
 
 TestedExpressions _findPreviousTestedExpressions(BinaryExpression node) {
-  Block block = node
-      .getAncestor((a) => a is Block && a.parent is BlockFunctionBody);
-  Iterable<AstNode> nodesInDFS = DartTypeUtilities.traverseNodesInDFS(block);
+  Block block =
+      node.getAncestor((a) => a is Block && a.parent is BlockFunctionBody);
```

Comment:
Added this code to the test and is not linted, but I am guessing that is a missing case for the rule, since it is the opposite bug I won't address it in this CL, also it fits with an improvement I have been thinking for this rule.